test/hotspot/jtreg/serviceability/jvmti/RedefineClasses/MissedStackMapFrames/MissedStackMapFrames.java

```diff
@@ -0,0 +1,124 @@
+ * @library /test/lib
+ *
+ * @run main/othervm/native -agentlib:MissedStackMapFrames MissedStackMapFrames
+ */
+
+import jdk.internal.org.objectweb.asm.ClassReader;
+import jdk.internal.org.objectweb.asm.ClassVisitor;
+import jdk.internal.org.objectweb.asm.MethodVisitor;
+import jdk.internal.org.objectweb.asm.Opcodes;
+
+
+public class MissedStackMapFrames {
+    static {
+        System.loadLibrary("MissedStackMapFrames");
+    }
+
+    /* For each test class:
+     *  - loads class (JNIEnv::FindClass);
+     *  - retransforms class (jvmtiEnv::RetransformClasses).
+     * Saves class bytes passed to ClassFileLoadHook.
+     */
+    private static native boolean doTest();
+
+    /* methods to analyze doTest results */
+    private static native int testCount();
+    private static native Class testClass(int idx);
+    private static native byte[] loadBytes(int idx);
+    private static native byte[] retransformBytes(int idx);
+
+    private static int getStackMapFrameCount(byte[] classfileBuffer) {
+        ClassReader reader = new ClassReader(classfileBuffer);
+        final int[] frameCount = {0};
+        ClassVisitor cv = new ClassVisitor(Opcodes.ASM9) {
+            @Override
+            public MethodVisitor visitMethod(int access, String name,
+                                             String descriptor, String signature,
+                                             String[] exceptions) {
+                return new MethodVisitor(Opcodes.ASM9) {
+                    private int methodFrames = 0;
+                    @Override
+                    public void visitFrame(int type, int numLocal, Object[] local,
+                                           int numStack, Object[] stack) {
+                        methodFrames++;
+                    }
+                    @Override
+                    public void visitEnd() {
+                        log("  method " + name + " - " + methodFrames + " frames");
+                        frameCount[0] += methodFrames;
+                    }
+                };
+            }
+        };
+        reader.accept(cv, 0);
+        return frameCount[0];
+    }
+
+    private static int checkStackMapFrames(String mode, byte[] classfileBuffer) {
+        log(mode + ", len = " + classfileBuffer.length);
+        int frameCount = getStackMapFrameCount(classfileBuffer);
+        log("  Has stack map frames: " + frameCount);
+        if (frameCount == 0) {
+            throw new RuntimeException(mode + " - no stack frames");
+        }
+        return frameCount;
+    }
+
+    private static void checkStackMapFrames(String mode, byte[] classfileBuffer, int expectedCount) {
+        int actualCount = checkStackMapFrames(mode, classfileBuffer);
+        if (actualCount != expectedCount) {
+            throw new RuntimeException(mode + " - unexpected stack frames count: " + actualCount
+                                       + " (expected " + expectedCount + ")");
+        }
+    }
```

Comment:
You added this method but are not using it. I think below the first call to checkStackMapFrames() needs to save the frameCount returned and then pass it in to the second call to checkStackMapFrames().